prov/hook/trace/src/hook_trace.c

```diff
@@ -831,6 +837,101 @@ static struct fi_ops_tagged trace_tagged_ops = {
 };
 
 
+static int trace_setname(fid_t fid, void *addr, size_t addrlen)
+{
+	int ret = 0;
+	struct hook_fabric *myfabric = hook_to_fabric(fid);
+
+	ret = fi_setname(hook_to_hfid(fid), addr, addrlen);
+	TRACE_CM(ret, myfabric, (addr) ? (*(uint64_t *)addr) : 0);
+
+	return ret;
+}
+
+static int trace_getname(fid_t fid, void *addr, size_t *addrlen)
+{
+	int ret = 0;
+	struct hook_fabric *myfabric = hook_to_fabric(fid);
+
+	ret = fi_getname(hook_to_hfid(fid), addr, addrlen);
+	TRACE_CM(ret, myfabric, (addr) ? (*(uint64_t *)addr) : 0);
+
+	return ret;
+}
+
+static int trace_getpeer(struct fid_ep *ep, void *addr, size_t *addrlen)
+{
+	int ret = 0;
+	struct hook_ep *myep = container_of(ep, struct hook_ep, ep);
+
+	ret = fi_getpeer(myep->hep, addr, addrlen);
+	TRACE_CM(ret, myep->domain->fabric, (addr) ? (*(uint64_t *)addr) : 0);
+
+	return ret;
+}
+
+static int trace_connect(struct fid_ep *ep, const void *addr,
+            const void *param, size_t paramlen)
+{
+	int ret = 0;
+	struct hook_ep *myep = container_of(ep, struct hook_ep, ep);
+
+	ret = fi_connect(myep->hep, addr, param, paramlen);
+	TRACE_CM(ret, myep->domain->fabric, (addr) ? (*(uint64_t *)addr) : 0);
+
+	return ret;
+}
+
+static int trace_listen(struct fid_pep *pep)
+{
+	struct hook_pep *mypep = container_of(pep, struct hook_pep, pep);
+
+    return fi_listen(mypep->hpep);
```

Comment:
Any reason these calls (listen/accept/reject/shutdown/join) are not traced? A single `TRACE_CM` might not fit all of them but you can always call `FI_TRACE` directly.

Reply:
Is there any useful information we want to log for listen/accept/reject/shutdown/join ops?  These ops only have pointers to ep/pep.  Will such information be useful for debugging?

Reply:
Indeed, the details are carried in `param`, but those are provider specific so not much can be done other than printing the address and length, which may not be very useful. However, merely info of these functions being called can sometimes be useful.

Reply:
indentation above is off.  I agree with @j-xiong , just knowing the calls were made can be useful.  Printing the ep pointers helps.